lib/vdsm/storage/exception.py

```diff
@@ -1349,6 +1349,29 @@ class VolumeGroupDoesNotExist(StorageException):
     code = 506
     msg = "Volume Group does not exist"
 
+    def __init__(self, vg_name='', vg_uuid='', error=''):
+        self.vg_name = vg_name
+        self.vg_uuid = vg_uuid
+        self.error = error
+
+    def _log_vg(self):
+        if self.vg_name:
+            return f"vg_name={self.vg_name}"
+
+        return f"vg_uuid={self.vg_uuid}"
+
+    @property
+    def value(self):
+        return f"{self._log_vg()} error={self.error}"
+
+    @classmethod
+    def from_vg_name(cls, vg_name, error=''):
+        return cls(vg_name=vg_name, error=error)
+
+    @classmethod
+    def from_vg_uuid(cls, vg_uuid):
+        return cls(vg_uuid=vg_uuid)
```

Comment:
The idea was to discourage the direct use of the constructor, because with all parameters optional it could be as well be misused and constructed without any parameter.

It still can, but at least the API hints how to use it. I was pondering different designs... One alternative would be to
```python
    assert vg_name or vg_uuid
```
... in the constructor of the exception. Or log an error if we don't want to raise from within an exception.